Zend/zend_alloc.c

```diff
@@ -1010,9 +1010,9 @@ static void *zend_mm_alloc_pages(zend_mm_heap *heap, uint32_t pages_count ZEND_F
 						goto get_chunk;
 					} else if (heap->overflow == 0) {
 #if ZEND_DEBUG
-						zend_mm_safe_error(heap, "Allowed memory size of %zu bytes exhausted at %s:%d (tried to allocate %zu bytes)", heap->limit, __zend_filename, __zend_lineno, size);
+						zend_mm_safe_error(heap, "Allowed memory size of %zu bytes exhausted at %s:%d (tried to allocate %zu bytes)", heap->limit, __zend_filename, __zend_lineno, ZEND_MM_CHUNK_SIZE);
 #else
-						zend_mm_safe_error(heap, "Allowed memory size of %zu bytes exhausted (tried to allocate %zu bytes)", heap->limit, ZEND_MM_PAGE_SIZE * pages_count);
+						zend_mm_safe_error(heap, "Allowed memory size of %zu bytes exhausted (tried to allocate %zu bytes)", heap->limit, ZEND_MM_CHUNK_SIZE);
 #endif
```

Comment:
@bwoebi Do you happen to know if it's correct that these two numbers already differ depending on ZEND_DEBUG (i.e. `size` vs `ZEND_MM_PAGE_SIZE * pages_count`)?

Reply:
I don't think this makes sense. "Tried to allocate %zu bytes" just won't give any info except of constant chunk size.

Script requests "size" bytes, MM doesn't find free pages and needs a new chunk.